File: 1010-pairs-of-songs-with-total-durations-divisible-by-60/1010-pairs-of-songs-with-total-durations-divisible-by-60.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& times) {
        
        unordered_map<int,int> umap;
        int count = 0;
        for(int time : times){
            if(umap.count(60-time%60)){
                count += umap[60-time%60];
            }
            if(time%60 !=0){
                umap[time%60]++;
            }else{
                umap[time%60+60]++;
            }
        }
        
        return count;
        
    }
};

File: 1010-pairs-of-songs-with-total-durations-divisible-by-60/1010-pairs-of-songs-with-total-durations-divisible-by-60.py
class Solution(object):
    def numPairsDivisibleBy60(self, time):
        ans, cnt = 0, [0] * 60
        for t in time: cnt[t % 60] += 1
        for i in range(1, 30): ans += cnt[i] * cnt[60 - i]
        return ans + cnt[0] * (cnt[0] - 1) // 2 + cnt[30] * (cnt[30] - 1) // 2
